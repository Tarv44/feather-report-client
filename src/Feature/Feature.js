import React, { Component } from 'react';
import styles from './Feature.module.css';

export default class Feature extends Component {
    static defaultProps = {
        selected: 'Select Feature'
    }

    state = {
        creating_new: false,
        new_feature: ''
    }

    updateCreatingNew(e) {
        e.preventDefault()
        this.setState({ creating_new: true})
    }

    updateFeature(f) {
        this.setState({ new_feature: f })
    }

    submitFeature(e) {
        e.preventDefault()
        this.props.updateCatFeatures(this.state.new_feature)
        this.props.updateProductFeatures(this.state.new_feature, this.props.index)
        this.setState({ creating_new: false })
    }

    render() {
        const FeatureOptions = this.props.features.map((f, i) => {
            return <option key={i}>{f.title}</option>
        })

        const Feature = this.state.creating_new
            ? (
                <div className={styles.feature}>
                    <input 
                        onChange={e => this.updateFeature(e.target.value, this.props.index)} 
                        type='text' 
                    />
                    <button 
                        className={styles.create} 
                        onClick={e => this.submitFeature(e, e.target.value)}
                    >
                        Submit Feature
                    </button>
                    <button 
                        onClick={e => this.props.removeFeature(e, this.props.index)}
                    >
                        Remove Feature
                    </button>
                </div>
            )
            : (
                <div className={styles.feature}>
                    <select 
                        value={this.props.selected}
                        onChange={e => this.props.updateFeature(e.target.value, this.props.index)} 
                    >
                        <option>Select Feature</option>
                        {FeatureOptions}
                    </select>
                    <button 
                        className={styles.create} 
                        onClick={e => this.updateCreatingNew(e)}
                    >
                        Create New Feature
                    </button>
                    <button 
                        onClick={e => this.props.removeFeature(e, this.props.index)}
                    >
                        Remove Feature
                    </button>
                </div>
            )

        return Feature
    }
}