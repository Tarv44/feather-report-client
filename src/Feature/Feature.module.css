.feature {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    border-bottom: solid black 1px;
    padding: .25em 0 .5em;
}

.create {
    margin-bottom: .25em;
}

