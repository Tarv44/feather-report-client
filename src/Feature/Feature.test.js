import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import Feature from './Feature'
import { features } from '../component.fixtures'

describe('----- Feature -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
            <Feature
                features={features}
            />
        , div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(<Feature
                features={features}
            />)
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})