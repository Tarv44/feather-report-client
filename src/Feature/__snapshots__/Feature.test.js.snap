// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- Feature ----- renders the UI as expected 1`] = `
<div
  className="feature"
>
  <select
    onChange={[Function]}
    value="Select Feature"
  >
    <option>
      Select Feature
    </option>
    <option>
      8-inch
    </option>
    <option>
      Ergonomic Handle
    </option>
    <option>
      Carbon Stainless Steel
    </option>
    <option>
      Waved Pattern
    </option>
    <option>
      Pakkawood Handle
    </option>
    <option>
      7-inch
    </option>
    <option>
      Black
    </option>
    <option>
      5-inch
    </option>
    <option>
      Lifetime Warranty
    </option>
    <option>
      Santoprene Handle
    </option>
    <option>
      Titanium Plated
    </option>
    <option>
      Comes with Sharpener
    </option>
    <option>
      Fiery Color
    </option>
  </select>
  <button
    className="create"
    onClick={[Function]}
  >
    Create New Feature
  </button>
  <button
    onClick={[Function]}
  >
    Remove Feature
  </button>
</div>
`;
